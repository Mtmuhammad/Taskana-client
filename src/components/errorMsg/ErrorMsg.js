// Component to show error msg for CRUD operations
import React from 'react'

const ErrorMsg = ({ errMsg, setErrMsg }) => {
  return (
    <div
      data-testid="error-container"
      className="msg d-flex justify-content-between alert alert-danger alert-dismissible fade show"
      role="alert"
    >
      <p
        data-testid="error-msg"
        style={{ fontSize: "1.3rem", marginBottom: "0" }}
      >
        <strong>Error!</strong> {errMsg}
      </p>
      <i
        data-testid="error-msg-icon"
        type="button"
        className="close text-danger bx bx-x"
        data-dismiss="alert"
        aria-label="Close"
        style={{ fontSize: "2.3rem" }}
        onClick={() => setErrMsg("")}
      ></i>
    </div>
  );
};

export default ErrorMsg;
