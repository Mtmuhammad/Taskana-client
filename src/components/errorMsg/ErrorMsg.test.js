import React from "react";
import { render, screen } from "@testing-library/react";
import ErrorMsg from "./ErrorMsg";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<ErrorMsg />);
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<ErrorMsg />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("error msg", () => {
  test("should render error msg container", () => {
    render(<ErrorMsg />);
    const container = screen.getByTestId("error-container");
    expect(container).toBeInTheDocument();
    expect(container).toHaveClass(
      "msg d-flex justify-content-between alert alert-danger alert-dismissible fade show"
    );
  });
  test("should render error msg", () => {
    render(<ErrorMsg />);
    const msg = screen.getByTestId("error-msg");
    expect(msg).toBeInTheDocument();
  });
  test("should render error msg icon", () => {
    render(<ErrorMsg />);
    const icon = screen.getByTestId("error-msg-icon");
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveClass("close text-danger bx bx-x");
  });
}); 
