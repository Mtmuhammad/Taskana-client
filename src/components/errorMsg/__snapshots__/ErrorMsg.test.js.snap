// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="msg d-flex justify-content-between alert alert-danger alert-dismissible fade show"
    data-testid="error-container"
    role="alert"
  >
    <p
      data-testid="error-msg"
      style="font-size: 1.3rem; margin-bottom: 0px;"
    >
      <strong>
        Error!
      </strong>
       
    </p>
    <i
      aria-label="Close"
      class="close text-danger bx bx-x"
      data-dismiss="alert"
      data-testid="error-msg-icon"
      style="font-size: 2.3rem;"
      type="button"
    />
  </div>
</DocumentFragment>
`;
